### PR TITLE
e2e: overwrite use caa tag when run e2e test

### PR DESCRIPTION
**Pull request by huoqifeng**
Fixes: #783

- Moved kustomize function from file `provision_ibmcloud.go` to `provision_ibmcloud_kustomize.go`
- Added property CAA_IMAGE_TAG so that it can be used to overwrite `latest` caa image tag for s390x
- Retrieve the commit hash if the e2e test is running from `staging` branch and no `CAA_IMAGE_TAG` provided

### CODE REVIEW

**Review by liudalibj — APPROVED** [review]
LGTM Thanks @huoqifeng

File: test/provisioner/provision_ibmcloud_kustomize.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+func (lio *IBMCloudInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config, properties map[string]string) error {
+	log.Debugf("%+v", properties)
+	var err error
+
+	// image
+	var newTag string
+	if IBMCloudProps.CaaImageTag != "" {
+		newTag = IBMCloudProps.CaaImageTag
+	} else {
+		newTag = getCaaNewTagFromCommit()
```
Comment (stevenhorsman) [inline test/provisioner/provision_ibmcloud_kustomize.go:142]: I guess a question here is whether we want to use the dev image (and therefore `latest` tag) for `amd64` and just the release build for `s390x`, or this behaviour of the release tag for all ibm cloud architectures?

I think that the current behaviour, of using release for both is fine, and I don't think there should be a big difference between them (but I I'm not an expert on this), but I'm just aware that it makes us different from the other cloud providers, so wanted to at least ask the question

Comment (huoqifeng) [reply test/provisioner/provision_ibmcloud_kustomize.go:138]: Right, maybe check `WORKER_FLAVOR` further is better.

**Review by stevenhorsman — APPROVED** [review]
This looks good to me. Thanks for the updated.

I've run this against an s390x cluster with no `CAA_TAG` in the properties and the tests worked:
```
Wait for the pod cc-operator-daemon-install-zphkn be ready
Wait for the kata runtimeclass be created
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created
time="2023-04-25T04:59:31-07:00" level=info msg="PodVM name: simple-peer-pod"
time="2023-04-25T04:59:31-07:00" level=debug msg="Instance number: 0, Instance id: 0797_03b6e432-e3e7-4ce7-a312-e5e4cf4850e4, Instance name: podvm-simple-peer-pod-3b7b3b74"
--- PASS: TestCreateSimplePod (40.36s)
    --- PASS: TestCreateSimplePod/Simple_Peer_Pod (40.36s)
        --- PASS: TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (0.22s)
=== RUN   TestCreatePodWithConfigMap
=== RUN   TestCreatePodWithConfigMap/Configmap_Pod
=== RUN   TestCreatePodWithConfigMap/Configmap_Pod/Configmap_is_created_and_contains_data
time="2023-04-25T05:00:17-07:00" level=info msg="Data Inside Configmap: Hello, world"
time="2023-04-25T05:00:17-07:00" level=info msg="Deleting pod... configmap-pod"
time="2023-04-25T05:00:17-07:00" level=info msg="Deleting Configmap... nginx-config"
--- PASS: TestCreatePodWithConfigMap (45.62s)
    --- PASS: TestCreatePodWithConfigMap/Configmap_Pod (45.62s)
        --- PASS: TestCreatePodWithConfigMap/Configmap_Pod/Configmap_is_created_and_contains_data (0.35s)
=== RUN   TestCreatePodWithSecret
=== RUN   TestCreatePodWithSecret/Secret_Pod
=== RUN   TestCreatePodWithSecret/Secret_Pod/Secret_is_created_and_contains_data
time="2023-04-25T05:00:53-07:00" level=info msg="Username inside volume: admin"
time="2023-04-25T05:00:53-07:00" level=info msg="Password inside volume: 123456"
time="2023-04-25T05:00:53-07:00" level=info msg="Deleting pod... secret-pod"
time="2023-04-25T05:00:53-07:00" level=info msg="Deleting Secret... nginx-secret"
--- PASS: TestCreatePodWithSecret (35.91s)
    --- PASS: TestCreatePodWithSecret/Secret_Pod (35.91s)
        --- PASS: TestCreatePodWithSecret/Secret_Pod/Secret_is_created_and_contains_data (0.67s)
PASS
ok  	github.com/confidential-containers/cloud-api-adaptor/test/e2e	193.065s
```
and the daemonset what created from the correct image:
```
Name:             cloud-api-adaptor-daemonset-h4bqd
Namespace:        confidential-containers-system
...
Containers:
  cloud-api-adaptor-con:
    Container ID:  containerd://42f02cc6b9f4796137d0597c1a07ec3e88dbc4c3f81ba53b3f50a67f48cd0652
    Image:         quay.io/confidential-containers/cloud-api-adaptor:1764961c2ae1e59704832bc148094df9f248a61e
    Image ID:      quay.io/confidential-containers/cloud-api-adaptor@sha256:db34c1e6986746301f98f3b11ca3747f390dcf96a33c12335899e1173a999a67
```
I then tried setting `CAA_IMAGE_TAG="fb83fdb36d1899f822b7423729357a9298775968"` and re-running the tests. They passed again, but this time the caa-ds picked up the image tag that I'd specified:
```
Name:             cloud-api-adaptor-daemonset-dbxxf
Namespace:        confidential-containers-system
Priority:         0
...
    Image:         quay.io/confidential-containers/cloud-api-adaptor:fb83fdb36d1899f822b7423729357a9298775968
```

The refactor into a separate kustomize.go is also nice.

Thanks QiFeng!

**Pull request merged**